src/ptolemy/raster.py

```diff
@@ -599,3 +605,163 @@ def raster_to_df(
+    index: pd.Index
+
+    @cached_property
+    def cell_area(self):
+        return xr.DataArray.from_series(cell_area_from_file(self.indicator)).astype(
+            dtype=self.boundary, copy=False
+        )
+
+    @property
+    def dim(self):
+        return self.index.name
+
+    @property
+    def spatial_dims(self):
+        return self.indicator.dims
+
+    @classmethod
+    def from_weighted_raster(
+        cls,
+        idxraster: xr.DataArray,
+        dim: Optional[str] = None,
+        allow_dask: bool = False,
+    ):
+        spatial_dims = ["lat", "lon"]
+        if dim is None:
+            non_spatial_dims = [d for d in idxraster.dims if d not in spatial_dims]
+            if len(non_spatial_dims) != 1:
+                raise ValueError(
+                    'Dimensions of idxraster should be ("lat", "lon", dim)'
+                )
+            dim = non_spatial_dims[0]
+
+        index = idxraster.indexes[dim].rename(dim)
+
+        idxraster = idxraster.assign_coords(
+            {dim: pd.RangeIndex(start=1, stop=len(index) + 1)}
+        )
+        is_boundary = lambda da: ((da > 0) & (da < 1)).any(dim)
+
+        idxr_stacked = idxraster.stack(spatial=spatial_dims)
+        boundary = idxr_stacked.sel(spatial=is_boundary(idxr_stacked))
+
+        within_ncountries = (idxraster >= 1).sum(dim)
+        if (within_ncountries > 1).any():
+            raise ValueError(
+                "Idxraster contains cells fully part of more than one {dim}"
+            )
+        indicator = (
+            (idxraster >= 1)
+            .idxmax(dim)
+            .where(within_ncountries & ~is_boundary(idxraster), 0)
+        )
+
+        if not allow_dask:
+            indicator = indicator.load()
+            boundary = boundary.load()
+
+        return cls(indicator=indicator, boundary=boundary, index=index)
+
+    @classmethod
+    def from_netcdf(cls, path, chunks=None):
+        ds = decode_compress_to_multi_index(xr.open_dataset(path, chunks=chunks))
+        name = ds.attrs["dim_name"]
+        spatial_dims = ds["indicator"].dims
+        index = ds.indexes[name]
+        boundary = (
+            ds["boundary"]
+            .rename({f"{n}_r": n for n in spatial_dims})
+            .assign_coords({name: pd.RangeIndex(1, len(index) + 1)})
+        )
+
+        return cls(
+            indicator=ds["indicator"].rename(name),
+            boundary=boundary,
+            index=index,
+        )
+
+    def to_netcdf(self, path):
+        boundary = self.boundary.rename(
+            {n: f"{n}_r" for n in self.spatial_dims}
+        ).assign_coords({self.dim: self.index})
+        xr.Dataset(
+            dict(
+                indicator=self.indicator,
+                boundary=boundary,
+            ),
+            attrs=dict(dim_name=self.dim),
+        ).pipe(encode_multi_index_as_compress).to_netcdf(path)
+
+    def aggregate(self, ndraster: xr.DataArray, func: str = "sum") -> xr.DataArray:
+        if func != "sum":
```

Comment:
This is being done, already, and is why i naively added `func` in the first place.

What currently happens if you use `mean` is:
1. the mean on the interior is computed (with `xarray_reduce(..., func=func)`)
2. the mean on the boundary is computed (with this getattr construct)
3. then they are **added** together

instead of 3), we would need to average them with some sort of number of cell counts weights.

or instead compute the sums of both, sum then and then divide by the number of cells.

Have not thought about this too much yet.

Reply:
What we could trivially do to be helpful nevertheless is add a an `interior_only=False` optional argument, which means that you want to compute whatever complicated statistic you are interested in only on the interior/indicator part.

Reply:
Nice, that sounds like a good addition! Basically then we would support lossy aggregations other than sum.